### DISCUSSION TITLE
Can not clone repo using ssh keys

### DISCUSSION THREAD

**Discussion by 3verness** [community/community#136171] [New to GitHub]
When I try to clone a repo in powershell, it shows:
```
git clone git@github.com:pyocd/libusb-package.git
Cloning into 'libusb-package'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
I have followed https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey to check my problem. But everything seem fine.
```
ssh-add -l -E sha256
4096 SHA256:SAME-AS-GITHUB-SETTINGS 3verness@gmail.com (RSA)
```
```
ssh -T git@github.com
Hi 3verness! You've successfully authenticated, but GitHub does not provide shell access.
```
I am using vscode terminal on windows:
```
 set GIT_SSH_COMMAND="ssh -v"
git clone git@github.com:pyocd/libusb-package.git
Cloning into 'libusb-package'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by SjxSubham** [comment]
Sometimes VS code gives U error. to resolve Use git bash as u know It's powerfull...

#coz sometimes VS code terminal Works As normal Terminal not like windows powershell

Using git bash to run any git command Will not let u get an error...!!!

**Comment by CapedBojji** [comment]
Hey i am having the same problem does anyone have a fix

**Comment by ghostinhershell** [comment]
Thanks for posting in the GitHub Community, @3verness!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the New to GitHub category is solely related to conversations around getting started on using GitHub and the community for beginners of GitHub, Community on Discussions, and those new to different areas of the product alike! This question should be in the General category. I've gone ahead and moved it for you. Good luck!

**Comment by KevalXD** [comment]
Even though your SSH key works (the ssh -T git@github.com proves it), Git sometimes can't find it when cloning.
The quickest thing to try first: Use HTTPS instead and see if it works.
powershellgit clone https://github.com/pyocd/libusb-package.git

If that clones fine, your SSH setup needs tweaking. If it fails too, you probably don't have access to that repo.

If you want to fix SSH properly, create a config file at `C:\Users\YourName\.ssh\config` and add:

Host github.com
  User git
  IdentityFile ~/.ssh/id_rsa
Then try cloning again with SSH. That usually does it.
One more thing to check: Run ssh -v git@github.com and look for a line saying your key is being offered. If it's not offering the right key, update the IdentityFile path above.
Let me know if HTTPS works or if you need help with the config file!

**Comment by Sonra0** [comment]
The issue is that your SSH authentication works fine (confirmed by ssh -T git@github.com), but there may be a conflict with which SSH client Git is using on Windows.
Quick Fix — Clone with HTTPS
For public repos, the easiest solution is to use HTTPS instead:
bashgit clone https://github.com/pyocd/libusb-package.git

If You Want SSH to Work
The likely culprit is the set syntax — that's CMD syntax and doesn't work in PowerShell. Use this instead:
powershell$env:GIT_SSH_COMMAND = "ssh -v"
git clone git@github.com:pyocd/libusb-package.git

Permanent Fix for SSH Consistency
Git for Windows sometimes uses its own bundled SSH client instead of Windows OpenSSH, which won't see your loaded keys. To force Git to always use Windows OpenSSH:
powershell[System.Environment]::SetEnvironmentVariable("GIT_SSH", "C:\Windows\System32\OpenSSH\ssh.exe", "User")
Then restart your terminal and try cloning via SSH again.


Why this happens: Even though ssh -T git@github.com succeeds (using Windows OpenSSH + your loaded key), Git may fall back to its own SSH client, which doesn't have access to your key — causing the permission denial.
